[김건우]Add: Monsters 과제(기능구현)

kgwoo opened the pull request:
## 수정 사항 간략한 한줄 요약
(여기에 수정 사항에 대한 간략한 한줄 요약/제목 작성해 주세요.)
- self-closing tag 적용

## 수정 사항들 자세한 내용
(여기에 수정 사항에 대한 자세한 내용을 작성해 주세요.)
- Monsters.js 파일의 SearchBox 와 CardList 태그에 self-closing Tag 적용		 

## 기타 질문 및 특이 사항
(궁금한 사항들, 하면서 느낀 점들 공유해주실 것이 있으면 자유롭게 작성해 주세요.)
- 감사합니다!


## 체크 리스트 (아래 사항들이 전부 체크되어야만 merge가 됩니다!)
- [X] 필요한 test들을 완료하였고 기능이 제대로 실행되는지 확인 하였습니다.
- [X] Wecode의 코드 스타일 가이드에 맞추어 코드를 작성 하였습니다.
- [X] 제가 의도한 파일들과 수정 사항들만 커밋이 된 것을 확인 하였습니다.
- [ ] 본 수정 사항들을 팀원들과 사전에 상의하였고 팀원들 모두 해당 PR에 대하여 알고 있습니다.

jamessoun93 commented on src/2-monsters/index.js:
```diff
@@ -27,17 +27,39 @@ class App extends Component {
 
   render() {
     // 필터링 로직
+    const filterMonster = this.state.monsters.filter((monster) => {
+      return monster.name.toLowerCase().includes(this.state.userInput)
+    })
```
비교대상인 monster.name 과 userInput 을 둘다 소문자 형태로 변환해주셔야 더욱 더 정확한 비교대상이 됩니다!

kgwoo committed:
Add: Monsters 과제(기능구현)
- jsonplaceholder의 API호출
- Card에 props로 id name email 전달
- map()사용하여, card를 cardlist에 배열만큼 전달
- filter() 사용하여, monsters 객체의 name과 input창의 value 값을 비교하여 일치하는 것을 띄워줌

jamessoun93 commented on src/Assignments/Monsters/students/11th/kgwoo/Components/Card/Card.js:
```diff
@@ -20,16 +20,16 @@ import "./Card.scss";
-          src="https://live.staticflickr.com/4561/38054606355_26429c884f_b.jpg"
+          src={`https://robohash.org/${this.props.id}?set=set2&size=180x180`}
           alt=""
         />
-        <h2>Name</h2>
-        <p>Email</p>
+        <h2>{this.props.name}</h2>
+        <p>{this.props.email}</p>
```
Goood 😎

jamessoun93 commented on src/Assignments/Monsters/students/11th/kgwoo/Components/CardList/CardList.js:
```diff
@@ -11,8 +11,14 @@ import "./CardList.scss";
+          return (
+            <Card key={monsters.id} id={monsters.id} name={monsters.name} email={monsters.email} />
+          );})}
+    </div>
```
`map()` 함수 사용 잘하셨네요! 🤩

jamessoun93 commented on src/Assignments/Monsters/students/11th/kgwoo/Monsters.js:
```diff
@@ -23,22 +23,31 @@ import "./Monsters.scss";
+    this.setState({userInput: e.target.value});
+  }
   render() {
+    let result = this.state.monsters.filter((monsters)=>{
+      return monsters.name.toLowerCase().includes(this.state.userInput.toLowerCase()); 
+    })
```
👏👏👏

jamessoun93 commented on src/Assignments/Monsters/students/11th/kgwoo/Monsters.js:
```diff
@@ -23,22 +23,31 @@ import "./Monsters.scss";
         <h1>컴포넌트 재사용 연습!</h1>
-        {/* <SearchBox handleChange=정의한메소드 /> */}
-        {/* <CardList monsters=몬스터리스트 /> */}
+        <SearchBox handleChange={this.handleChange}></SearchBox>        
+        <CardList  monsters={result}></CardList>
```
React 에서는 JSX tag 내에 자식 요소 혹은 text 가 없다면 self-closing tag 를 활용하여 해당 태그를 닫아줄 수 있죠? 

수정해주세요!

jamessoun93 reviewed (changes requested):
전반적으로 너무 좋습니다!

코멘트 확인해서 요청사항 반영해서 다시 push 해주세요!

kgwoo committed:
Refactor: self-closing Tag 적용
- monsters.js 파일에 SearchBox 와 CardList에 sel-closing tag 적용

jamessoun93 commented:
LGTM! 수고하셨습니다! 🙇🏻‍♂️

kgwoo commented:
> LGTM! 수고하셨습니다! 🙇🏻‍♂️
감사합니다!!
